### PR TITLE
Add get_item_schema function to WP_REST_Widget_Areas_Controller

### LINKED ISSUE #15902: Release 5.8.0 introduced PHP Warnings, at least in 'wp shell'
After upgrading to v5.8.0 I noticed when working in wp shell that I always get two of these warnings when the shell starts up:

Warning: Invalid argument supplied for foreach() in /home/ourpreschoollife/public_html/wp-includes/rest-api/endpoints/class-wp-rest-controller.php on line 287

Reverting to 5.7.9 removes them.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
Fixes: https://github.com/WordPress/gutenberg/issues/15902

This PR implements the get_item_schema function for WP_REST_Widget_Areas_Controller class.

This fixes a problem where warnings were being displayed during wp-cli usage.
Props to @n9yty, @TimothyBJacobs, and @lkraav for providing valuable information that allowed this fix.

## How has this been tested?
I executed wp CLI shell again a WordPress instance running Gutenberg master. Using wp shell --path=/MY/WP/PATH
I verified no error were thrown on master I got:
Warning: Invalid argument supplied for foreach() in /Users/pc/dev/core/wordpres-develop/build/wp-includes/rest-api/endpoints/class-wp-rest-controller.php on line 287
Warning: Invalid argument supplied for foreach() in /Users/pc/dev/core/wordpres-develop/build/wp-includes/rest-api/endpoints/class-wp-rest-controller.php on line 287

### CODE REVIEW

File: lib/class-wp-rest-widget-areas-controller.php [hunk 1]
```diff
@@ -84,6 +84,53 @@ public function register_routes() {
...
+	 * Retrieves the comment's schema, conforming to JSON Schema.
+	 *
+	 * @since 4.7.0
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'Widget Area',
```
Comment (TimothyBJacobs) [inline lib/class-wp-rest-widget-areas-controller.php]:
```suggestion
			'title'      => 'widget-area',
```

WordPress deviates from the JSON Schema spec here and uses it more like the `id` field. See the block renderer controller.

File: lib/class-wp-rest-widget-areas-controller.php [hunk 2]
```diff
@@ -84,6 +84,53 @@ public function register_routes() {
...
+							'context'     => array( 'view', 'edit', 'embed' ),
+						),
+						'rendered'      => array(
+							'description' => __( 'HTML content for the object, transformed for display.' ),
+							'type'        => 'string',
+							'context'     => array( 'view', 'edit', 'embed' ),
+							'readonly'    => true,
+						),
+						'block_version' => array(
+							'description' => __( 'the current version of the block format that the content string is usin' ),
```
Comment (TimothyBJacobs) [inline lib/class-wp-rest-widget-areas-controller.php]:
```suggestion
							'description' => __( 'Version of the content block format used by the object.' ),
```

I think this should probably match the existing posts controller.

**Review by TimothyBJacobs — DISMISSED** [review]
Since the endpoint now has a schema, the args should ideally be provided by `::get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE )` when registering the `update_item` route.

**Comment by aduth** [comment]
@TimothyBJacobs Would you be able to give a fresh review here after the latest changes?

**Comment by TimothyBJacobs** [comment]
The schema looks good to me, but the `register_rest_route` call should use that schema for the `$args` option when registering the edit route. This can be done by calling `$this→get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE )`.

**Comment by jorgefilipecosta** [comment]
Hi @TimothyBJacobs thank you for the feedback your suggestion was applied 👍

**Comment by TimothyBJacobs** [comment]
@jorgefilipecosta I think there is an issue with the `update_item` method call. The code only accounts for when `content` is a `string`, but it should also check whether it is an `object` with the `raw` property. 

```
$sidebar_content = $request->get_param( 'content' );
```

An example from the posts controller.

```
if ( is_string( $request['content'] ) ) {
	$prepared_post->post_content = $request['content'];
} elseif ( isset( $request['content']['raw'] ) ) {
	$prepared_post->post_content = $request['content']['raw'];
}
```

**Comment by TimothyBJacobs** [comment]
@jorgefilipecosta Do you want me to push the changes up for this?

**Comment by jorgefilipecosta** [comment]
Hi @TimothyBJacobs sorry for the delay I had some AFK days, I will soon update this PR.

**Comment by jorgefilipecosta** [comment]
Hi @TimothyBJacobs, thank you for the review. The issue identified was fixed.
We now handle the raw property for content.
It is possible to test this using the following code:
```
wp.apiFetch({
    path: '/__experimental/widget-areas/sidebar-1',
    data: { content: '<!-- wp:columns --><div class="wp-block-columns has-2-columns"><!-- wp:column --><div class="wp-block-column"><!-- wp:paragraph --><p>A</p><!-- /wp:paragraph --></div><!-- /wp:column --><!-- wp:column --><div class="wp-block-column"><!-- wp:paragraph --><p>B</p><!-- /wp:paragraph --></div><!-- /wp:column --></div><!-- /wp:columns -->' },
    method: 'POST',
}).then(console.log);
```

**Review by TimothyBJacobs — APPROVED** [review]
Looks good to me!

**Pull request merged**